backend/pkg/controllers/main.controller.go

```diff
@@ -681,6 +680,69 @@ func (c *MainController) RefreshTokenHandler(ctx *gin.Context) {
 	})
 }
 
+func (c *MainController) AuthenticateAgent(ctx *gin.Context) {
+	var user models.User
+	userId, err := getUserIdFromToken(ctx)
+	if err != nil {
+		ctx.JSON(http.StatusUnauthorized, gin.H{"error": err.Error()})
+		return
+	}
+
+	result := c.usersCollection.FindOne(ctx, bson.M{"userId": userId})
+	err = result.Decode(&user)
+	if err != nil {
+		fmt.Printf("Error: %s", err)
+		ctx.JSON(400, gin.H{"error": err.Error()})
+		return
+	}
+	if user.AgentBearerToken != "" {
+		ctx.JSON(http.StatusOK, gin.H{
+			"message": "Success",
+			"token":   user.AgentBearerToken,
+		})
+		return
+	}
+
+	token, err := createToken(userId, user.UserName, "")
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	_, err = c.usersCollection.UpdateOne(ctx,
+		bson.M{"userId": userId},
+		bson.M{"$set": bson.M{
+			"agentBearerToken": token,
+		},
+		},
+	)
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	ctx.JSON(http.StatusOK, gin.H{
+		"message": "Success",
+		"token":   token,
+	})
+}
+
+func (c *MainController) CheckAgentAuthorization(ctx *gin.Context) {
```

Comment:
as this is a middleware please move it to the middleware directory

Reply:
Cannot do, I need access to main controller to query database, I don't know any workaround to that